### ISSUE TITLE
Where are the Standard ML communities and enthusiasts?

### ISSUE THREAD

**Issue by kytrinyx** [exercism/sml#97]
As we move towards the launch of the new version of Exercism we are going to be ramping up on actively recruiting people to help provide feedback.

Our goal is to get to 100%: everyone who submits a solution and wants feedback should get feedback. Good feedback. You can read more about this aspect of the new site here: http://mentoring.exercism.io/

To do this, we're going to need a lot more information about where we can find language enthusiasts.

* Is Standard ML supported by one or more large organizations?
* Does Standard ML have an official community manager?
* Do you know of specific communities (online or offline) that are enthusiastic about Standard ML? (Chat communities, forums, meetups, student clubs, etc)
* Are there popular conferences for Standard ML? (If so, what are some examples?)
* Are there any organizations who are targeted specifically at getting certain subgroups or demographics interested in Standard ML? (e.g. kids, teenagers, career changers, people belonging to various groups that are typically underrepresented in tech?)
* Are there specific groups or programs dedicated to mentoring people in Standard ML?
* Are there popular newsletters for Standard ML?
* Is Standard ML taught at programming bootcamps? (If so, what are some examples?)
* Is Standard ML taught at universities? (If so, what are some examples?)

In other words: where do people care a lot and/or know a lot about Standard ML?

This is part of the project being tracked in https://github.com/exercism/meta/issues/103

**Comment by sshine** [comment]
Hello @kytrinyx!
Cc @snahor

I've signed up as mentor on this track.

TL;DR: The only place on the Internet where I've found Standard ML to be alive is on StackOverflow. There's an [abundance of people](https://stackoverflow.com/tags/sml/topusers) willing to answer the very few questions that get asked. I don't expect the need for a lot of mentors, but there might be ones willing there.

Since I'm currently mentoring on the Haskell, Ocaml and Standard ML tracks, I'd like to become the maintainer of this track and bring the exercises up-to-date compared to the Haskell track. (The Ocaml track is similarly a little outdated since Steve is apparently the only active mentor there.)

Also, thanks a lot for making this platform available! :-)

----

I'll answer your questions from my perspective:

- *Is Standard ML supported by one or more large organizations?*

  **No.** Standard ML is used almost entirely in academia. I personally know of two commercial SML programmers, one in finance and one in computer graphics. My guess is that there's <25 in the world.

- *Does Standard ML have an official community manager?*

  **No.** The community is fragmented.

- *Do you know of specific communities (online or offline) that are enthusiastic about Standard ML?*

  **Yes.** Besides the small communities that surround the compilers MLton, SML/NJ, Moscow ML and Poly/ML (of which I think only MLton and Poly/ML are truly alive), there's the #sml channel on Freenode IRC and the very low-traffic StackOverflow [`sml` tag](https://stackoverflow.com/tags/sml).

- *Are there popular conferences for Standard ML?*

  **No.** There's [ICFP](https://www.icfpconference.org/), [POPL](https://conf.researchr.org/series/POPL) and the like, but none specific to Standard ML.

- *Are there any organizations who are targeted specifically at getting certain subgroups or demographics interested in Standard ML?*

  **No.**

- *Are there popular newsletters for Standard ML?*

  **No.**

- *Is Standard ML taught at programming bootcamps?*

  **No.** Not that I know of. Haskell and Ocaml are more "real-world" and often get picked in favour.

- *Is Standard ML taught at universities?*

  **Yes.** That's the only thing Standard ML is used for. Examples: [Coursera's Programming Languages](https://www.coursera.org/learn/programming-languages). The University of Copenhagen used Standard ML for more than a decade but recently switched to F#. The maintainer of [Moscow ML](http://mosml.org/) is a professor there. Replace the university, the compiler and the maintainer and you get similar responses for the compilers mentioned.

  Considering the massive decline in SML questions on StackOverflow over the past two years, I'm guessing that no introductory courses use SML any more. There used to be seasonal influx, but that's gone.

**Comment by kytrinyx** [comment]
This is excellent, thank you!

> Since I'm currently mentoring on the Haskell, Ocaml and Standard ML tracks, I'd like to become a maintainer of this track and bring the exercises up-to-date compared to the Haskell track. (The Ocaml track is similarly a little outdated since Steve Bastians is apparently the only active mentor there.)

Brilliant! I've invited you to be a maintainer on this repository, and if you'd also like to be added to OCaml, I'd gladly add you there as well.

**Comment by sshine** [comment]
Sure, I'd like that.

**Comment by mcmillhj** [comment]
@kytrinyx Since we are doing role updates, would it be possible to remove me as a maintainer? I just haven't had the time lately to add beneficial contributions.

**Comment by kytrinyx** [comment]
@mcmillhj Indeed! I'll do this now. (Do you want to submit a PR that sets yourself as `alumni` in https://github.com/exercism/sml/blob/master/config/maintainers.json?)

**Comment by mcmillhj** [comment]
@kytrinyx definitely! I'll do that now.

**Comment by sshine** [comment]
@mcmillhj: Thanks for making the language track so pleasant to take over!

I'll close this issue as there isn't a mentor bottleneck for this track.
